Fails to build with Qt 6.10

### Checklist

- [x] I **have searched** the issue tracker and **was unable** to find an [open](../issues?q=is%3Aissue+is%3Aopen) or [closed](../issues?q=is%3Aissue+is%3Aclosed) issue matching what I'm seeing.
- [x] I **have verified** that the issue is happening in the **[latest nightly build](https://ci.86box.net/job/86Box/lastSuccessfulBuild/artifact/)**, or the latest commit, if building from source.

### What happened?

Fails to build with Qt 6.10 because access to private components was changed.  Needs changes to cmake scripts. 
```
/home/builder/build/src/86box-git/src/qt/qt_mainwindow.cpp:127:10: fatal error: qpa/qplatformwindow.h: No such file or directory
  127 | #include <qpa/qplatformwindow.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

### Configuration file

```ini
N/A
```

### Operating system

Arch Linux

### 86Box version

74678a391b8806661b18a7fd9251b26d3b887fc0

### Build architecture

Linux - x64 (64-bit)

### Build type

- [ ] New recompiler
- [ ] Debug build

### Additional context

Error occurs during build, so runtime info not relevant.

--- a/src/qt/CMakeLists.txt
+++ b/src/qt/CMakeLists.txt
@@ -48,10 +48,12 @@ set(CMAKE_FIND_PACKAGE_PREFER_CONFIG OFF)
 find_package(Threads REQUIRED)
 find_package(Qt${QT_MAJOR} COMPONENTS Core Widgets Network OpenGL Gui REQUIRED)
 find_package(Qt${QT_MAJOR}LinguistTools REQUIRED NO_CMAKE_FIND_ROOT_PATH)
-if(NOT USE_QT6)
-    # For <qpa/qplatformwindow.h>  in src/qt/qt_mainwindow.cpp
+if(USE_QT6)
+    set(QT_NO_PRIVATE_MODULE_WARNING ON)
+    find_package(Qt6 COMPONENTS GuiPrivate)
+else()
     include_directories(${Qt5Gui_PRIVATE_INCLUDE_DIRS})
-endif ()
+endif()
 
 # TODO: Is this the correct way to do this, and is it required on any
 # other platforms or with Qt 5?

